src/components/v2/Layout/Header/CoinInfoContainers.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { useVaiUser } from 'hooks/useVaiUser';
+import { useMarketsUser } from 'hooks/useMarketsUser';
+import { CoinInfo } from './CoinInfo';
+
+export const XvsCoinInfo = () => {
+  const { userVaiBalance } = useVaiUser();
+  return <CoinInfo balance={userVaiBalance} coin="vai" />;
+};
+
+export const VaiCoinInfo = () => {
+  const { userXvsBalance } = useMarketsUser();
+  return <CoinInfo balance={userXvsBalance} coin="xvs" />;
+};
```

Comment:
That looks good. But that's still an non-pure component inside the `components` folder which is meant to host only pure components 😂
Here's how I see this: if we want to keep following this logic of separating pure components from containers, then this file just needs to be moved to the `containers` folder and we're good to go.

If you want to have containers next to components directly, that's ok for me too, but in this case then we should get rid of the `containers` folder as then there's no more distinction made between pure components and containers.
That's the last thing for me.

It's up to you to decide on that, I'll go with the option your prefer. Everything else I see in this PR is great and I agree with 👍